docs/TypeDenotation.md

```diff
@@ -0,0 +1,53 @@
+# Type Denotation
+
+Type Denotation is used to describe semantic information around what the inputs and outputs are.    It is stored on the TypeProto message.
+
+## Motivation
+
+The motivation of such a mechanism can be illustrated via a simple example. In the the neural network SqueezeNet, it takes in an NCHW image input float[1,3,244,244] and produces a output float[1,1000,1,1]:
+
+```
+input_in_NCHW -> data_0 -> SqueezeNet() -> output_softmaxout_1
+```
+
+In order to run this model the user needs a lot of information.    In this case the user needs to know:
+* the input is an image
+* the image is in the format of NCHW
+* the color channels are in the order of bgr
+* the pixel data is 8 bit
+* the pixel data is normalized as values 0-255
+
+This proposal consists of three key components to provide all of this information: 
+* Type Denotation, 
+* [Dimension Denotation](DimensionDenotation.md), 
+* [Model Metadata](MetadataProps.md).
+
+## Type Denotation Definition
+
+To begin with, we define a set of semantic types that define what models generally consume as inputs and produce as outputs.
+
+Specifically, in our first proposal we define the following set of standard denotations:
+
+1. `IMAGE` describes that a type holds an image.  You can use dimension denotation to learn more about the layout of the image, and also the optional model metadata_props.
+2. `AUDIO` describes that a type holds an audio clip.   
+3. `TEXT` describes that a type holds a block of text.
+
```

Comment:
What about good old numerical tensors, for other tasks such as recommendations, forecasting, anomaly detection, etc?
Should we add another type TENSOR for those?

Reply:
Makes sense !    I added the 3 we had been using the most in our models (image/audio/text) and added explicit metadata for image.   I assumed that there would be follow up proposals and PR's as we add more "types" here.    What you thinking ?     Also, we do have the normal case where there is no denotation .  In that case the tensor is also a good old numerical tensor.     since denotation is optional, it still has all the fun stuff, like shape and type.